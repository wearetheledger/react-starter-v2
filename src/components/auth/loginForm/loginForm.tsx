import { requiredRule } from "@utils/rules/requiredRule";
import { Button, Form, Icon, Input } from "antd";
import { WrappedFormUtils } from "antd/lib/form/Form";
import * as React from "react";
import { FormattedMessage, InjectedIntl } from "react-intl";

export interface LoginFormProps {
	form: WrappedFormUtils;
	intl: InjectedIntl;
	handleSubmit(e: React.FormEvent): void;
}

export interface LoginFromValues {
	userName: string;
	password: string;
}

export const LoginForm: React.StatelessComponent<LoginFormProps> = props => {
	const {
		form: { getFieldDecorator },
		handleSubmit,
		intl
	} = props;

	return (
		<Form onSubmit={handleSubmit} className="login-form">
			<Form.Item>
				{getFieldDecorator("userName", {
					rules: [requiredRule]
				})(
					<Input
						size="large"
						prefix={<Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />}
						// tslint:disable-next-line: no-any
						placeholder={intl.formatMessage({ id: "auth.username" })}
					/>
				)}
			</Form.Item>
			<Form.Item>
				{getFieldDecorator("password", {
					rules: [requiredRule]
				})(
					<Input
						size="large"
						prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
						type="password"
						placeholder={intl.formatMessage({ id: "auth.password" })}
					/>
				)}
			</Form.Item>
			<Form.Item>
				<Button size="large" type="primary" htmlType="submit" block={true}>
					<FormattedMessage id="auth.login" />
				</Button>
			</Form.Item>
		</Form>
	);
};
