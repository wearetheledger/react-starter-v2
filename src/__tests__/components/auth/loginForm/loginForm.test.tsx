import { LoginForm } from "@components/auth/loginForm/loginForm";
import React from "react";
import { createRendererWithProvider } from "__tests__/helpers/createRendererWithProvider";
import { mockForm } from "__tests__/helpers/mockForm";
import { mockIntl } from "__tests__/helpers/mockIntl";

it("should match snapshot", () => {
	const wrapper = createRendererWithProvider(
		<LoginForm intl={mockIntl} form={mockForm} handleSubmit={jest.fn()} />
	)();
	const tree = wrapper.toJSON();
	expect(tree).toMatchSnapshot();
});
