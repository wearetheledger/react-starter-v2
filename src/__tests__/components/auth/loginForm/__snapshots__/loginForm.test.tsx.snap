// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<form
  className="ant-form ant-form-horizontal login-form"
  onSubmit={[MockFunction]}
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-col ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          >
            <span
              className="ant-input-prefix"
            >
              <i
                aria-label="icon: user"
                className="anticon anticon-user"
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </i>
            </span>
            <input
              className="ant-input ant-input-lg"
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              size="large"
              style={null}
              type="text"
              value=""
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-col ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
          >
            <span
              className="ant-input-prefix"
            >
              <i
                aria-label="icon: lock"
                className="anticon anticon-lock"
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="lock"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                  />
                </svg>
              </i>
            </span>
            <input
              className="ant-input ant-input-lg"
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              size="large"
              style={null}
              type="password"
              value=""
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-col ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
            onClick={[Function]}
            type="submit"
          />
        </span>
      </div>
    </div>
  </div>
</form>
`;
